src/diagnostics.rs

```diff
@@ -239,10 +239,12 @@ fn verilator_syntax(
 
     static RE: std::sync::OnceLock<Regex> = std::sync::OnceLock::new();
     let re = RE.get_or_init(|| {
-        Regex::new(
-            r"%(?P<severity>Error|Warning)(-(?P<warning_type>[A-Z0-9_]+))?: [^:]+:(?P<line>\d+):((?P<col>\d+):)? ?(?P<message>.*)",
-        )
-        .unwrap()
+        let file_path_str = regex::escape(file_path.to_str().unwrap_or(""));
+        let pattern = format!(
+            r"%(?P<severity>Error|Warning)(-(?P<warning_type>[A-Z0-9_]+))?: {}:(?P<line>\d+):((?P<col>\d+):)? ?(?P<message>.*)",
+            file_path_str
```

Comment:
The regex is created inside a oncelock to ensure that we only compile it once. So this change would bake in the file path of the first file that was checked by verilator.

Reply:
That explains why after these changes the diagnostics only worked on the first file I opened and not the follow-up ones :) But I was not able to find out why given that as I said I am not familiar with rust :)